cmd/cored/main.go

```diff
@@ -131,6 +132,61 @@ func runServer() {
 	ctx := context.Background()
 	env.Parse()
 
+	// We need to be able to add handlers to our serve mux in two phases.
+	// In the first phase, we will start
+	// listening on the raft routes (`/raft`). This allows us to do things like
+	// read the config value stored in raft storage. (A new node in a raft cluster
+	// can't read values without kicking off a consensus round, which in turn
+	// requires this node to be listening for raft requests.)
+	//
+	// Once this node is able to read the config value, it can set up the remaining
+	// cored functionality, and add the rest of the core routes to the serve mux.
+	// That is the second phase.
+	mux := http.NewServeMux()
+
+	var handler http.Handler = mux
+	handler = reqid.Handler(handler)
+
+	secureheader.DefaultConfig.PermitClearLoopback = true
+	secureheader.DefaultConfig.HTTPSRedirect = httpsRedirect
+	secureheader.DefaultConfig.Next = handler
+
+	server := &http.Server{
+		Addr:         *listenAddr,
+		Handler:      secureheader.DefaultConfig,
+		ReadTimeout:  httpReadTimeout,
+		WriteTimeout: httpWriteTimeout,
+		// Disable HTTP/2 for now until the Go implementation is more stable.
+		// https://github.com/golang/go/issues/16450
+		// https://github.com/golang/go/issues/17071
+		TLSNextProto: map[string]func(*http.Server, *tls.Conn, http.Handler){},
+	}
+
+	// The `ListenAndServe` call has to happen in its own goroutine because
```

Comment:
Nit: personally I would avoid markdown-style backticks in code comments like this, since nothing will ever render them as markdown and the font is already monospace. To me it's more readable without them. But not a big deal.